### PR TITLE
As a developer I want to access redis DB with authentication

### PR DESCRIPTION
**Pull request by AngithCJ**
**Changes**
- Redis password authentication enabled in Redis server deployment
- Redis DB connection updated to use the password in lib persistent manager
- Added logic to read the encrypted password from the deployment file and use 
it for Redis.
- Unit test updated
- Redis password authentication for GitHub.

Note: The changes in the PR include for the stories BRUCE-5355 and BRUCE-5875

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]

File: install/Docker/dockerfiles/Dockerfile.redis [hunk 1]
```diff
@@ -23,7 +23,8 @@ RUN if [ -z "$ODIMRA_USER_ID" ] || [ -z "$ODIMRA_GROUP_ID" ]; then \
     && groupadd -r -g $ODIMRA_GROUP_ID odimra \
     && useradd -s /bin/bash -u $ODIMRA_USER_ID -m -d /home/odimra -r -g odimra odimra
 
-RUN apt-get install bash sed
+RUN apt-get -y update
```
Comment (shivacharanms) [inline install/Docker/dockerfiles/Dockerfile.redis:26]: Do not update explicitly

File: install/Docker/dockerfiles/Dockerfile.redis [hunk 2]
```diff
@@ -23,7 +23,8 @@ RUN if [ -z "$ODIMRA_USER_ID" ] || [ -z "$ODIMRA_GROUP_ID" ]; then \
...
+RUN apt-get -y install build-essential libssl-dev openssl bash sed
```
Comment (shivacharanms) [inline install/Docker/dockerfiles/Dockerfile.redis:27]: What is the need for these packages? explain the need for each

File: install/Docker/dockerfiles/scripts/redis-checkdb.sh [hunk 3]
```diff
@@ -16,14 +16,8 @@
 # Script is for generating certificate and private key
 # for Client mode connection usage only
 sleep 3
-if [[ -n ${REDIS_DEFAULT_PASSWORD} ]]; then
 echo "Checking if default entries already present"
-redis-cli -a ${REDIS_DEFAULT_PASSWORD} -h ${master} -p ${REDIS_HA_REDIS_SERVICE_PORT} --tls --cert ${TLS_CERT_FILE} --key ${TLS_KEY_FILE} --cacert ${TLS_CA_CERT_FILE} <<HERE
+redis_password=$(openssl pkeyutl -decrypt -in cipher -inkey ${ODIMRA_RSA_PRIVATE_FILE} -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha256)
```
Comment (shivacharanms) [inline install/Docker/dockerfiles/scripts/redis-checkdb.sh:20]: sha512 needs to be used. change across all references in PR

File: lib-utilities/config/config_test.go [hunk 4]
```diff
@@ -68,7 +134,9 @@ func TestSetConfiguration(t *testing.T) {
                 "OnDiskHost": "localhost",
                 "OnDiskPort": "6380",
                 "MaxIdleConns": 10,
-                "MaxActiveConns": 120
+                "MaxActiveConns": 120,
+				"RedisInMemoryPasswordFilePath": "/tmp/redis_inmemory_password",
```
Comment (shivacharanms) [inline lib-utilities/config/config_test.go:138]: Indentation needs to be fixed

**Review by shivacharanms — CHANGES_REQUESTED** [review]

File: install/Docker/dockerfiles/Dockerfile.redis (hunk 1 above)
Comment (AngithCJ) [reply install/Docker/dockerfiles/Dockerfile.redis:26]: Ok

File: install/Docker/dockerfiles/Dockerfile.redis (hunk 2 above)
Comment (AngithCJ) [reply install/Docker/dockerfiles/Dockerfile.redis:27]: Need to install openssl, and build-essential and libssl-dev are dependency packages for that

File: install/Docker/dockerfiles/Dockerfile.redis (hunk 1 above)
Comment (AngithCJ) [reply install/Docker/dockerfiles/Dockerfile.redis:26]: As I cross checked, system can not locate openssl command without update command and build fail with the error 'can not locate openssl'

File: install/Docker/dockerfiles/Dockerfile.redis (hunk 2 above)
Comment (AngithCJ) [reply install/Docker/dockerfiles/Dockerfile.redis:27]: This was mistake from my side, apparently build essential and libssl-dev is not needed for openssl. I will remove this

**Review by amar-shalgar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**